### PR TITLE
Add setup.py back to fix cross compiling ARM64

### PR DESCRIPTION
**Pull request by sontek**
ARM64 builds are broken currently because `poetry` does not support platform specific builds currently.  The issues for it are here:

The issues that relate to this problem are:
 https://github.com/python-poetry/poetry/issues/2051
 https://github.com/python-poetry/poetry/issues/2613

with the code for the bug it has here:
https://github.com/python-poetry/poetry-core/blob/main/src/poetry/core/masonry/builders/wheel.py#L298-L312

Basically `cibuildwheel` is sending a couple of environment variables that `setuptools` uses normally to decide platform/arch/etc. 

```
        - _PYTHON_HOST_PLATFORM
        - ARCHFLAGS
```

These are *not* respected by poetry and so if you need to cross-compile libraries `poetry` is currently not the build tool for this.   I think we could look at https://github.com/pypa/hatch as an alternative or send a PR to poetry to fix this.  But for now this gets us back using `setup.py` so cross compiling works.

### CODE REVIEW

File: Pipfile.lock [hunk 1]
```diff
@@ -0,0 +1,534 @@
+{
```
Comment (kapilt) [inline Pipfile.lock:1]: why re pipfile/pipfile.lock

Comment (kapilt) [reply Pipfile.lock:1]: setup.py and requirements.txt seems sufficient, rather than duplicating requirements.

**Comment by kapilt** [comment]
could you add a description on what the intent of this pr is

File: Pipfile.lock (hunk 1 above)
Comment (sontek) [reply Pipfile.lock:1]: This was to provide the same ergonomics as `poetry`, where we can `pipenv run ...` and `pip env lock` our dev dependencies and have the whole chain locked so we don't get random upgrades during CI/CD that break anything. 

I used the `Pipfile` to generate the `requirements-dev.txt` file, so it isn't a duplicate, just included for convience.

I'm not tied to `pipenv` or anything like that.   This is a library so its not going to have strict dependency requirements, I was just trying to match the experience we had with `poetry`

**Comment by sontek** [comment]
@kapilt Description added

File: Pipfile.lock (hunk 1 above)
Comment (sontek) [reply Pipfile.lock]: I removed pipenv

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Comment by sontek** [comment]
I don't have access to merge so you'll have to do it

**Pull request merged**